src/coins.cpp

```diff
@@ -110,9 +117,9 @@ CCoinsMap::const_iterator CCoinsViewCache::FetchCoins(const uint256 &txid) const
 }
 
 
-bool CCoinsViewCache::GetAnchorAt(const uint256 &rt, ZCIncrementalMerkleTree &tree) const {
-    CAnchorsMap::const_iterator it = cacheAnchors.find(rt);
-    if (it != cacheAnchors.end()) {
+bool CCoinsViewCache::GetSproutAnchorAt(const uint256 &rt, ZCIncrementalMerkleTree &tree) const {
+    CAnchorsSproutMap::const_iterator it = cacheSproutAnchors.find(rt);
```

Comment:
Could this be done with a template method and a switch statement similar to AbstractPopAnchor(...) and PopAnchor(...) to avoid having the two methods GetSproutAnchorAt and GetSaplingAnchorAt?

Reply:
It can't because they're virtual functions. :(